controllers/kuadrant_controller.go

```diff
@@ -370,14 +402,47 @@ func (r *KuadrantReconciler) reconcileAuthorino(ctx context.Context, kObj *kuadr
 		},
 	}
 
-	err := r.SetOwnerReference(kObj, authorino)
+	err = r.SetOwnerReference(kObj, authorino)
 	if err != nil {
 		return err
 	}
 
 	return r.ReconcileResource(ctx, &authorinov1beta1.Authorino{}, authorino, reconcilers.CreateOnlyMutator)
 }
 
+// TODO(eguzki): Define better whether istio is enabled or not
+func (r *KuadrantReconciler) isIstioEnabled(ctx context.Context) (bool, error) {
+	iop := &iopv1alpha1.IstioOperator{}
+	iopKey := client.ObjectKey{Name: iopName(), Namespace: iopNamespace()}
+	if err := r.Client().Get(ctx, iopKey, iop); err != nil {
+		if apimachinerymeta.IsNoMatchError(err) {
+			return false, nil
+		}
+		if apierrors.IsNotFound(err) {
+			return true, nil
+		}
+
+		return false, err
+	}
+
+	return true, nil
+}
+
+// extAuthEnabled is how we find external auth enable flag in a context.Context.
+type extAuthEnabled struct{}
+
+func newContextWithIstioEnabled(ctx context.Context, istioEnabled bool) context.Context {
```

Comment:
The name of this function makes it look like that when invoked Istio _is_ enabled, while this is not always the case. Maybe `newContextWithIstioInfo`?

Reply:
done